### PR TITLE
DISK HAS TAGS

### PR DESCRIPTION
**Pull request by abdullahaslam306**

### CODE REVIEW

File: plugins/azure/virtualmachines/diskHasTags.js [hunk 1]
```diff
@@ -0,0 +1,49 @@
+var async = require('async');
+
+var helpers = require('../../../helpers/azure/');
+
+module.exports = {
+    title: 'Disk Has Tags',
```
Comment (alphadev4) [inline plugins/azure/virtualmachines/diskHasTags.js:6]:
```suggestion
    title: 'VM Disk Has Tags',
```

File: plugins/azure/virtualmachines/diskHasTags.js [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+    category: 'Virtual Machines',
+    domain: 'Compute',
+    description: 'Ensures that Azure virtual machine disks have tags.',
```
Comment (alphadev4) [inline plugins/azure/virtualmachines/diskHasTags.js:9]:
```suggestion
    description: 'Ensures that Azure virtual machine disks have tags associated.',
```

File: plugins/azure/virtualmachines/diskHasTags.js [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+    more_info: 'Tags help you to group resources together that are related to or associated with each other. It is a best practice to tag cloud resources to better organize and gain visibility into their usage.',
+    link: 'https://learn.microsoft.com/en-us/azure/azure-resource-manager/management/tag-resources',
+    recommended_action: 'Modify Disk and add tags.',
```
Comment (alphadev4) [inline plugins/azure/virtualmachines/diskHasTags.js:12]:
```suggestion
    recommended_action: 'Modify VM Disk and add tags.',
```

File: plugins/azure/virtualmachines/diskHasTags.js [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+            }
+            if (!disks.data.length) {
+                helpers.addResult(results, 0, 'No existing disk volumes found', location);
+                return rcb();
+            }
+            for (let disk of disks.data) {
+                if (!disk.id) continue;
+
+                if (disk.tags && Object.entries(disk.tags).length > 0){
+                    helpers.addResult(results, 0, 'Disk has tags associated', location, disk.id);
```
Comment (alphadev4) [inline plugins/azure/virtualmachines/diskHasTags.js:38]:
```suggestion
                    helpers.addResult(results, 0, 'VM disk has tags.', location, disk.id);
```

File: plugins/azure/virtualmachines/diskHasTags.js [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+                } else {
+                    helpers.addResult(results, 2, 'Disk does not have tags associated', location, disk.id);
```
Comment (alphadev4) [inline plugins/azure/virtualmachines/diskHasTags.js:40]:
```suggestion
                    helpers.addResult(results, 2, 'VM disk does not have tags', location, disk.id);
```

File: plugins/azure/virtualmachines/diskHasTags.js [hunk 6]
```diff
@@ -0,0 +1,49 @@
+var async = require('async');
+
+var helpers = require('../../../helpers/azure/');
+
+module.exports = {
+    title: 'VM Disk Has Tags',
+    category: 'Virtual Machines',
+    domain: 'Compute',
+    description: 'Ensures that Azure virtual machine disks have tags associated.',
```
Comment (alphadev4) [inline plugins/azure/virtualmachines/diskHasTags.js:9]:
```suggestion
    description: 'Ensure that Azure virtual machine disks have tags associated.',
```

File: plugins/azure/virtualmachines/diskHasTags.spec.js [hunk 7]
```diff
@@ -0,0 +1,88 @@
...
+                done();
+            });
+        });
+
+        it('should give failing result if Disk volume has BYOK encryption disabled', function(done) {
+            const cache = createCache([disks[1]]);
+            diskHasTags.run(cache, {}, (err, results) => {
+                expect(results.length).to.equal(1);
+                expect(results[0].status).to.equal(2);
+                expect(results[0].message).to.include('Disk does not have tags associated');
```
Comment (alphadev4) [inline plugins/azure/virtualmachines/diskHasTags.spec.js:82]:
```suggestion
                expect(results[0].message).to.include('VM disk does not have tags');
```

File: plugins/azure/virtualmachines/diskHasTags.spec.js [hunk 8]
```diff
@@ -0,0 +1,88 @@
...
+                done();
+            });
+        });
+
+        it('should give passing result if Disk volume has BYOK encryption enabled only', function(done) {
+            const cache = createCache([disks[0]]);
+            diskHasTags.run(cache, {}, (err, results) => {
+                expect(results.length).to.equal(1);
+                expect(results[0].status).to.equal(0);
+                expect(results[0].message).to.include('Disk has tags associated');
```
Comment (alphadev4) [inline plugins/azure/virtualmachines/diskHasTags.spec.js:71]:
```suggestion
                expect(results[0].message).to.include('VM disk has tags');
```

File: plugins/azure/virtualmachines/diskHasTags.js [hunk 9]
```diff
@@ -0,0 +1,49 @@
...
+            }
+            if (!disks.data.length) {
+                helpers.addResult(results, 0, 'No existing disk volumes found', location);
+                return rcb();
+            }
+            for (let disk of disks.data) {
+                if (!disk.id) continue;
+
+                if (disk.tags && Object.entries(disk.tags).length > 0){
+                    helpers.addResult(results, 0, 'VM disk has tags.', location, disk.id);
```
Comment (alphadev4) [inline plugins/azure/virtualmachines/diskHasTags.js:38]:
```suggestion
                    helpers.addResult(results, 0, 'VM disk has tags', location, disk.id);
```

File: plugins/azure/virtualmachines/diskHasTags.js [hunk 10]
```diff
@@ -0,0 +1,49 @@
+var async = require('async');
+
+var helpers = require('../../../helpers/azure/');
+
+module.exports = {
+    title: 'VM Disk Has Tags',
```
Comment (alphadev4) [inline plugins/azure/virtualmachines/diskHasTags.js:6]: do we need to change the file name as well?

**Review by AkhtarAmir — APPROVED** [review]

**Pull request merged**